Add script to generate coverage reports

dbatyai opened the pull request.

yichoi commented on tools/measure_coverage.sh:
```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+# Copyright 2016-present Samsung Electronics Co., Ltd. and other contributors
```
``2017-present``

nova0821 commented:
@dbatyai Hello, is there any description about this PR? (How to run the test, The result that generated with this script, etc.)

Also, if it has new option in ``check_test.js``, related documents should be updated.
``measure_coverage.sh`` also has some dependency(node.js and some modules), you need to update documents about it.

nova0821 commented:
@dbatyai  I installed all dependency modules, but it is failed to run ``measure_coverage.sh`` on my ubuntu 14.04

dbatyai commented:
@nova0821 Hi, I've added help/usage to the script, and updated it to be less dependent on directories. Also, I've updated the documentation where it was relevant.

Please check it again. If for some reason it still fails for you, then please provide a little detail about how you tried to run it, and the output.

nova0821 commented on tools/measure_coverage.sh:
```diff
@@ -0,0 +1,124 @@
+    echo "Please install all required dependencies"
+    echo ""
+    echo "    sudo apt install npm lcov nodejs gcc-multilib"
+    echo "    npm install --save-dev babel-cli nyc babel-plugin-istanbul"
```
I tried to run only ``sudo apt install npm lcov nodejs gcc-multilib`` command and run this script again, I got the error message.

```bash
nova@nova-VirtualBox:~/iotjs/coverage$ ./tools/measure_coverage.sh 
./tools/measure_coverage.sh: line 41: realpath: command not found
./tools/measure_coverage.sh: line 56: realpath: command not found
Could not find the node_modules directory, please specify it
as an argument.
```

It looks better to show ``npm install --save-dev babel-cli nyc babel-plugin-istanbul`` in this case.

Also, does this script have dependency ``realpath``?

dbatyai commented:
Tuns out istanbul and babel require node version > 4.0, and ubuntu 14.04 only comes with 0.10. I've updated the script to reflect this, you'll have to update your node package and reinstall nodejs.
Also I removed the realpath dependency.

nova0821 commented:
@dbatyai Ok, I'll check it again.

kolipka commented on tools/measure_coverage.sh:
```diff
@@ -0,0 +1,138 @@
+    echo ""
+    echo "Running the script will require some additional dependencies:"
+    echo "    sudo apt install npm lcov nodejs gcc-multilib"
+    echo "    npm install --save-dev babel-cli nyc babel-plugin-istanbul"
```
Why you save this dependency? I think it should be inside package.json, and you expect it to be already installed, or you install it without saving. You can perform npm install babel-cli in empty directory and it will work

kolipka commented on tools/measure_coverage.sh:
```diff
@@ -0,0 +1,138 @@
+    echo "Measure JavaScript and C coverage and create a html report"
+    echo "out of the results"
+    echo ""
+    echo "Usage: $0 [NODE_MODULES_DIR]"
```
why this path is required? You can assume that it is in main iotjs folder (so relative path from script will be ../node_modules)

knightburton replied:
This path is not required, it is optional. Basically we assume that it is in main iotjs folder.

kolipka commented on tools/measure_coverage.sh:
```diff
@@ -0,0 +1,138 @@
+dpkg -l nodejs >> /dev/null 2>&1 && \
+test "$(nodejs -v)" \> "v4.0.0"
+
+if [ "$?" -ne "0" ]; then
```
could you move help and dependency info to function, to avoid duplication?

kolipka commented on tools/measure_coverage.sh:
```diff
@@ -0,0 +1,138 @@
+cd $iotjs_root
+
+# Transpile JS files to provide line counters
+/usr/bin/nodejs $modules_dir/.bin/babel \
```
Are you sure this node is correct? You tested node from path for ver > 4.0. I could have old node installed in /usr/bin and new elsewhere with path pointing to it.

kolipka commented on tools/measure_coverage.sh:
```diff
@@ -0,0 +1,138 @@
+mv src/orig_js src/js
+
+# Generate js coverage report
+/usr/bin/nodejs $modules_dir/.bin/nyc report --reporter=lcov \
```
Again, coud be problem with node version

nova0821 commented:
@dbatyai I checked, it works with Node > 4.x. Could you check comments which @kolipka mentioned above?

yichoi commented:
@dbatyai any updates?

knightburton commented:
@kolipka Updated the PR following your comments.

kolipka commented:
@knightburton awesome, thanks!
